source/crud/read-operations/cursor.txt

```diff
@@ -0,0 +1,150 @@
+=========================
+Access Data From a Cursor
+=========================
+
+.. default-domain:: mongodb
+
+Read operations that return multiple documents do not immediately return
+all values matching the query. Because a query can potentially match
+very large sets of documents, these operations rely upon an
+object called a **cursor** that fetches documents in batches to reduce
+both memory consumption and network bandwidth usage. Cursors
+are highly configurable and offer multiple interaction paradigms for
+different use cases.
+
+The following functions directly return cursors:
+
+- ``Collection.find()``
+
+- ``Collection.aggregate()``
+
+- ``Collection.listIndexes()``
+
+- ``Db.aggregate()``
+
+- ``Db.listCollections()``
+
+Other methods, like ``Collection.findOne()`` and ``Collection.watch()``
+use cursors to generate results, but do not directly expose those
+cursors to the method caller.
+
+.. warning::
+
+   Do not combine different cusor paradigms on a single cursor.
+   Cursors are designed under the assumption that users will use only
+   **one** of the following paradigms to access data. Failure to use
+   just one of these paradigms will lead to undefined behavior. For
+   instance, calling both ``count()`` and ``toArray()`` or both
+   ``hasNext()`` and ``forEach()`` on the same cursor will likely lead
+   to an error.
+
+.. warning::
+
+   Do not execute multiple async methods on a single cursor in parallel.
+   Only perform one async operation at a time on a single cursor.
+   Attempting multiple async operations in parallel on a single cursor
+   will result in undefined behavior. For example,
+   pushing the result of multiple ``next()`` method calls to a
+   collection of promises and then calling ``Promise.all()`` on that
+   collection of promises will likely lead to an error.
+
+View Cursor Contents One By One
+-------------------------------
+
+Many use cases only require one document at a time. Cursors are designed
```

Comment:
Good point, I've renamed this section to make it more paradigm (though I'm going to avoid using that word) centric and I'm changing the setup to just talk about paradigms in general rather than talking about use cases at all.